UniversalMediaEngine.Tests/MediaEngineTests.cs

```diff
@@ -0,0 +1,533 @@
+            }
+            catch(Exception e)
+            {
+                exceptionThrown = true;
+            }
+
+            Assert.IsTrue(exceptionThrown);
+                
+        }
+
+        [TestMethod]
+        public async Task TestPlayWithInvalidUrlFails()
+        {
+            // Arrange
+            var engineUnderTest = new MediaEngine();
+            bool callbackOccurred = false;
+            uint callbackTimeout = 10;
+
+            // Act
+            var result = await engineUnderTest.InitializeAsync();
+
+            engineUnderTest.MediaStateChanged += ((MediaState newState) =>
+            {
+                // We're looking for the error state as this should be returns for
+                // an invalid URL
+                if (MediaState.Error == newState)
+                    callbackOccurred = true;
+            });
+
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
+
+            try
+            {
+                engineUnderTest.Play("");
+
+                // Wait for the callback to be invoked until the timeout expires
+                uint counter = 0;
+                while (!callbackOccurred)
+                {
+                    if (counter == callbackTimeout)
+                        Assert.IsFalse(true, "Timed out waiting for callback from mediaengine");
+
+                    await Task.Delay(100);
+
+                    counter++;
+                }
+            }
+            catch (Exception e)
+            {
+                Assert.Fail("An exception was thrown during execution of the test: " + e.Message);
+            }
+
+            // Assert
+
+            // The callback should be call within the timeout period
+            Assert.IsTrue(callbackOccurred);
+        }
+
+        [TestMethod]
+        public async Task TestPlayStreamWithInitializeSucceeds()
+        {
+            // Arrange
+            var engineUnderTest = new MediaEngine();
+            bool callbackOccurred = false;
+            uint callbackTimeout = 10;
+            StorageFile testFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///Assets/TestSilence.wav"));
+
+            // Act
+            var result = await engineUnderTest.InitializeAsync();
+
+            Assert.IsTrue(MediaEngineInitializationResult.Success == result);
+
+            engineUnderTest.MediaStateChanged += ((MediaState newState) =>
+            {
+                if (MediaState.Playing == newState)
+                    callbackOccurred = true;
+            });
+
+            try
+            {         
+                // Open the file as a byte stream
+                engineUnderTest.PlayStream(await testFile.OpenReadAsync());
+
+                // Wait for the callback to be invoked until the timeout expires
+                uint counter = 0;
+                while (!callbackOccurred)
+                {
+                    if (counter == callbackTimeout)
```

Comment:
Might this be cleaner as:

Assert.IsFalse(counter == callbackTimeout, "timed out...")

(same as earlier comment)